observatory-platform/observatory/platform/utils/release_utils.py

```diff
@@ -0,0 +1,449 @@
+    """Get a list of datasets for a telescope running for a given organisation name.
+    Assumes each organisation only runs 1 telescope of a given type. Only returns the first
+    1000 datasets for that telescope.
+
+    :param telescope_type_id: Telescope type ID.
+    :param org_name: Organisation name.
+    :return: List of Dataset objects.
+    """
+
+    dataset_limit = 1000
+
+    api = make_observatory_api()
+    telescope_type = api.get_telescope_type(type_id=telescope_type_id)
+    organisation_id = get_organisation_id(org_name)
+    telescopes_info = api.get_telescopes(1, telescope_type_id=telescope_type.id, organisation_id=organisation_id)
+    telescope_id = telescopes_info[0].id
+    datasets = api.get_datasets(telescope_id=telescope_id, limit=dataset_limit)
+    return datasets
+
+
+def get_dataset(*, telescope_type_id: str, org_name: str) -> "Dataset":  # noqa: F821
+    """Assuming there is only 1 dataset for a given Telescope, get the dataset.
+
+    :param telescope_type_id: Telescope type id, e.g., onix.
+    :param org_name: Organisation name of the associated telescope.
+    :return: Dataset id.
+    """
+
+    datasets = get_datasets(telescope_type_id=telescope_type_id, org_name=org_name)
+    return datasets[0]
+
+
+def get_dataset_id(*, telescope_type_id: str, org_name: str) -> int:
+    """Assuming there is only 1 dataset for a given Telescope, get its dataset id.
+
+    :param telescope_type_id: Telescope type id, e.g., onix.
+    :param org_name: Organisation name of the associated telescope.
+    :return: Dataset id.
+    """
+
+    dataset = get_dataset(telescope_type_id=telescope_type_id, org_name=org_name)
+    return dataset.id
+
+
+def find_dataset_id_from_name(*, name: str, datasets: List["Dataset"]) -> int:  # noqa: F821
+    """Get the dataset id for a given dataset name, from a list of Datasets.
+    If no dataset is found, it will throw an exception.
+
+    :param name: Name of the dataset to search for.
+    :param datasets: List of datasets to search.
+    :return: Dataset id.
+    """
+
+    for dataset in datasets:
+        if dataset.name == name:
+            return dataset.id
+
+    raise Exception(f"Dataset {name} not found")
+
+
+def find_dataset_id_from_address(*, address: str, datasets: List["Dataset"]) -> int:  # noqa: F821
+    """Get the dataset id for a given dataset storage address, from a list of Datasets.
+    If no dataset is found, it will throw an exception.
+
+    :param address: Storage address of the dataset to search for.
+    :param datasets: List of datasets to search.
+    :return: Dataset id.
+    """
+
+    limit = 1000
+
+    for dataset in datasets:
+        api = make_observatory_api()
+        storages = api.get_dataset_storages(dataset_id=dataset.id, limit=limit)
+        for storage in storages:
+            if storage.address == address:
+                return dataset.id
+
+    raise Exception(f"Dataset with address {address} not found")
```

Comment:
These functions seem like they are going to create brittle code because they are determining the dataset_id based on names and dataset addresses.